### PR TITLE
Support S3 IAD Regional option

### PR DESCRIPTION
**Pull request by mullermp**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mullermp** [comment]
backport of https://github.com/aws/aws-sdk-ruby/pull/2158/files

**Comment by mullermp** [comment]
```
AWS_REGION='us-east-1' cucumber -t @buckets
9 scenarios (9 passed)
39 steps (39 passed)
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ Unreleased Changes
 2.11.419 (2019-12-18)
 ------------------
 
+* Feature - Support `s3_us_east_1_regional_endpoint` from `SharedConfig`.
```
Comment (mullermp) [inline CHANGELOG.md]: Not sure if this should be a new version or not. Gem has not been published for htis version.

File: aws-sdk-core/lib/aws-sdk-core/plugins/s3_iad_regional_endpoint.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        host.sub(".us-east-1", '')
+      end
+
+      private
+
+      def self.resolve_iad_regional_endpoint(cfg)
+        mode = ENV['AWS_S3_US_EAST_1_REGIONAL_ENDPOINT'] ||
+          Aws.shared_config.s3_us_east_1_regional_endpoint(profile: cfg.profile) ||
+          'legacy'
+        unless %w(legacy regional).include?(mode)
```
Comment (trevorrowe) [inline aws-sdk-core/lib/aws-sdk-core/plugins/s3_iad_regional_endpoint.rb]: I see that we require the lower case "legacy" or "regional" mode. Would it make sense to downcase the mode first to deal with minor upper/lower case inconsistencies?

Comment (mullermp) [reply aws-sdk-core/lib/aws-sdk-core/plugins/s3_iad_regional_endpoint.rb]: I'm going with the harmless yet elegant `downcase!` option.

**Pull request merged**